opencti-platform/opencti-graphql/tests/02-integration/01-database/user-delete-cleanup-test.ts

```diff
@@ -0,0 +1,163 @@
+import { describe, expect, it } from 'vitest';
+import { ADMIN_USER } from '../../utils/testQuery';
+import { generateInternalId, generateStandardId } from '../../../src/schema/identifier';
+import { ENTITY_TYPE_USER } from '../../../src/schema/internalObject';
+import type { AuthContext, AuthUser } from '../../../src/types/user';
+import {
+  addNotification,
+  addTrigger, myNotificationsFind,
+  triggerGet
+} from '../../../src/modules/notification/notification-domain';
+import type {
+  MemberAccessInput,
+  TriggerLiveAddInput,
+  WorkspaceAddInput
+} from '../../../src/generated/graphql';
+import { addUser, assignGroupToUser, findById as findUserById, userDelete } from '../../../src/domain/user';
+import { ACCOUNT_STATUS_ACTIVE } from '../../../src/config/conf';
+import { ADMINISTRATOR_ROLE } from '../../../src/utils/access';
+import type { StoreMarkingDefinition } from '../../../src/types/store';
+import { addWorkspace, editAuthorizedMembers, findById as findWorkspaceById } from '../../../src/modules/workspace/workspace-domain';
+import type {
+  NotificationAddInput,
+} from '../../../src/modules/notification/notification-types';
+import { TriggerEventType, TriggerType } from '../../../src/generated/graphql';
+
+/**
+ * Create a new user in elastic for this test purpose using domain APIs only.
+ * @param adminContext
+ * @param adminUser
+ * @param username
+ */
+const createUserForTest = async (adminContext: AuthContext, adminUser: AuthUser, username: string) => {
+  const userToDeleteId: string = generateStandardId(ENTITY_TYPE_USER, { user_email: `${username}@opencti.io` });
+  // const userToDeleteInternalId: string = generateInternalId();
+
+  const simpleUser = {
+    id: userToDeleteId,
+    password: 'changeme',
+    user_email: `${username}@opencti.io`,
+    name: username,
+    firstname: username,
+    lastname: 'opencti'
+  };
+  const createdUser = await addUser(adminContext, adminUser, simpleUser);
+
+  // TODO what is the type here ??
+  return createdUser;
+};
+
+/**
+ * Create a AuthUser from user information that have been used to create the user in elastic.
+ * // TODO there is probably a better way to have a AuthUser from a User...
+ * @param user
```

Comment:
Maybe this isn't relevant, but have you tried using the createUser method in` utils/testQuery.ts`?

Reply:
I think thar createUser from testQuery uses graphQL requests , and since tests are domain & engine level only it's an issue because it could lead to unconsistency. That's why we rewrote a createUser in the file. But this domain level one could be in a common file. We can do that in a follow up PR.